rasa_core/server.py

```diff
@@ -868,6 +869,83 @@ async def retrieve_conversation_messages(request: Request, sender_id: Text):
+
+        Returns
+            responses: A jsonify dict of sender_ids on which the action trigger were call to
+        """
+        # retrieve parameters
+        request_params = request.json
+        action_to_execute = (request_params.get("name") or
+                             request_params.get("action"))
+        policy = request_params.get("policy", None)
+        confidence = request_params.get("confidence", None)
+        verbosity = event_verbosity_parameter(request,
+                                              EventVerbosity.AFTER_RESTART)
+        # dict of responses that will store all the ids that were updated as keys
+        responses = {}
+        # Retrieve ids from tracker_store
+        ids = retrieve_keys(app)
+        # For each ids, trigger an action if it was stalled for more than a day 
+        for id in ids:
+            tracker = app.agent.tracker_store.get_or_create_tracker(id)
+            id_state = tracker.current_state(verbosity)
+            # check if the last event was made within the last 86400secs (24h)
+            current_time = datetime.datetime.utcnow()
+            last_event_ts_from_id = datetime.datetime.utcfromtimestamp(id_state["latest_event_time"])
+            secondsinaday = 86400
+            if compare_utcdatetime_with_gap(current_time, last_event_ts_from_id, secondsinaday):
+                try:
+                    output_channel = _get_output_channel(request, tracker)
+                    logger.info('output_channel: {}'.format(output_channel))
+                    await app.agent.execute_action(id,
+                                                action_to_execute,
+                                                output_channel,
+                                                policy,
+                                                confidence)
+                    # add id to result
+                    responses[id] = "Action trigger sent"
+
+                except ValueError as e:
+                    raise ErrorResponse(400, "ValueError", e)
+                except Exception as e:
+                    logger.error("Encountered an exception while running action '{}'. "
+                                "Bot will continue, but the actions events are lost. "
+                                "Make sure to fix the exception in your custom "
+                                "code.".format(action_to_execute))
+                    logger.debug(e, exc_info=True)
+                    raise ErrorResponse(500, "ValueError",
+                                        "Server failure. Error: {}".format(e))
+        return response.json(responses)
+
+def retrieve_keys(app):
+    """ Retrieves the list of keys of the tracker store
+
+    Returns:
+        list -- tracker_store list of keys
+    """
+    keys = []
+    if app.agent.tracker_store:
+        keys = list(app.agent.tracker_store.keys())
+    return keys
+
+def compare_utcdatetime_with_gap(dt_a, dt_b, gap):
```

Comment:
One minor comment here, although using 'gap' as variable here reflects is to be a time gap (or time difference), but would be good to explicitly say that in variable name